lib/chef/provider/mount/linux.rb

```diff
@@ -53,6 +53,10 @@ def mounted?
             when %r{\A#{Regexp.escape(real_mount_point)}\s+([/\w])+\[#{device_mount_regex}\]\s}
               mounted = true
               logger.trace("Bind device #{device_logstring} mounted as #{real_mount_point}")
+            # Permalink for network device mounted to an existing mount point: https://rubular.com/r/HicK63SqchPLvk
+            when %r{\A#{Regexp.escape(real_mount_point)}\s+#{device_mount_regex}\[([\/\w])+\]\s}
```

Comment:
maybe just this?

`%r{\A#{Regexp.escape(real_mount_point)}\s+#{device_mount_regex}\[}`

does any of the stuff after the bracket really matter in terms of determining a match or not?

Reply:
the extra matching on the allowed characters within the brackets may actually be brittle as well, if there's ever a mountpoint in the world which doesn't match those characters there.

Reply:
Agreed, I have updated the Regex.